src/Functions/reverseDNSQuery.cpp

```diff
@@ -0,0 +1,128 @@
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnArray.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Common/DNSResolver.h>
+#include <Poco/Net/IPAddress.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+class ReverseDNSQuery : public IFunction
+{
+public:
+    static constexpr auto name = "reverseDNSQuery";
```

Comment:
Let's add some restrictions for this function and disable it by default. We can do it in a way similar to introspection functions (see `allow_introspection_functions` setting, and `dictGet`, `INTROSPECTION`, `addressToLine`, and `addressToSymbol` privileges in RBAC) or simply introduce a global config setting and check it like `Context::getGlobalContextInstance()->getConfigRef().getBool("allow_reverse_dns_query_function", false))` in `executeImpl`. A global setting will be enough for testing purposes.

Reply:
Should I also move the docs from `docs/sql-reference/functions/ip-address-functions.md` to `docs/sql-reference/functions/introspection.md`?

Reply:
No, `ip-address-functions.md` looks more suitable for this function

Reply:
Is this last option tweakable in SQL? I have added both stateless and integration tests, if this setting is not tweakable in SQL, it won't work for the stateless one

Reply:
> Is this last option tweakable in SQL?

No, it has to be set in config.xml

> if this setting is not tweakable in SQL, it won't work for the stateless one

See `./tests/config/config.d` and `./tests/config/install.sh`

Reply:
Neat, done